pom.xml

```diff
@@ -1394,9 +1394,9 @@
         <spotbugs.skip>false</spotbugs.skip>
         <spotbugs.onlyAnalyze>org.eclipse.jetty.*</spotbugs.onlyAnalyze>
       </properties>
-      <modules>
-        <module>javadoc</module>
-      </modules>
+<!--      <modules>-->
+<!--        <module>javadoc</module>-->
+<!--      </modules>-->
```

Comment:
why?

Reply:
I didn't make this change.

@olamy what's the reason for this?

Reply:
It appears that this was the solution to some kind of build issue (seen here)

https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/activity?branch=PR-5921

I'll dig into this.

Reply:
Looks like javadoc fails due to ...

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:03 h
[INFO] Finished at: 2021-01-27T02:13:51Z
[INFO] ------------------------------------------------------------------------
---------------------------------------------------
constituent[0]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/conf/logging/
constituent[1]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/jansi-1.17.1.jar
constituent[2]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-slf4j-provider-3.6.3.jar
constituent[3]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-transport-wagon-1.4.1.jar
constituent[4]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-connector-basic-1.4.1.jar
constituent[5]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/wagon-file-3.3.4.jar
constituent[6]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/jcl-over-slf4j-1.7.29.jar
constituent[7]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/jsoup-1.12.1.jar
constituent[8]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/wagon-http-3.3.4-shaded.jar
constituent[9]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/commons-cli-1.4.jar
constituent[10]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/cdi-api-1.0.jar
constituent[11]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/org.eclipse.sisu.plexus-0.3.4.jar
constituent[12]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/wagon-provider-api-3.3.4.jar
constituent[13]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/plexus-interpolation-1.25.jar
constituent[14]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-compat-3.6.3.jar
constituent[15]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/plexus-component-annotations-2.1.0.jar
constituent[16]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/org.eclipse.sisu.inject-0.3.4.jar
constituent[17]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-spi-1.4.1.jar
constituent[18]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-impl-1.4.1.jar
constituent[19]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-provider-3.6.3.jar
constituent[20]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-artifact-3.6.3.jar
constituent[21]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-repository-metadata-3.6.3.jar
constituent[22]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-core-3.6.3.jar
constituent[23]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/commons-lang3-3.8.1.jar
constituent[24]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/slf4j-api-1.7.29.jar
constituent[25]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/plexus-cipher-1.7.jar
constituent[26]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/plexus-sec-dispatcher-1.4.jar
constituent[27]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/plexus-utils-3.2.1.jar
constituent[28]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/jsr250-api-1.0.jar
constituent[29]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/javax.inject-1.jar
constituent[30]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/guava-25.1-android.jar
constituent[31]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/guice-4.2.1-no_aop.jar
constituent[32]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/commons-io-2.5.jar
constituent[33]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-shared-utils-3.2.1.jar
constituent[34]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-util-1.4.1.jar
constituent[35]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-resolver-api-1.4.1.jar
constituent[36]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-builder-support-3.6.3.jar
constituent[37]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-model-builder-3.6.3.jar
constituent[38]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-model-3.6.3.jar
constituent[39]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-plugin-api-3.6.3.jar
constituent[40]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-settings-builder-3.6.3.jar
constituent[41]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-settings-3.6.3.jar
constituent[42]: file:/home/jenkins/agent/tools/hudson.tasks.Maven_MavenInstallation/maven3/apache-maven-3.6.3/lib/maven-embedder-3.6.3.jar
---------------------------------------------------
Exception in thread "main" java.lang.NoSuchMethodError: 'org.apache.http.HttpHost org.apache.http.client.utils.URIUtils.extractHost(java.net.URI)'
	at org.eclipse.aether.transport.http.HttpTransporter.<init>(HttpTransporter.java:122)
	at org.eclipse.aether.transport.http.HttpTransporterFactory.newInstance(HttpTransporterFactory.java:71)
	at org.eclipse.aether.internal.impl.DefaultTransporterProvider.newTransporter(DefaultTransporterProvider.java:104)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.<init>(BasicRepositoryConnector.java:129)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory.newInstance(BasicRepositoryConnectorFactory.java:155)
	at org.eclipse.aether.internal.impl.DefaultRepositoryConnectorProvider.newRepositoryConnector(DefaultRepositoryConnectorProvider.java:107)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:496)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:401)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:207)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:240)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor(DefaultRepositorySystem.java:255)
	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve(DefaultPluginDependenciesResolver.java:106)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getPluginDescriptor(DefaultMavenPluginManager.java:182)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getMojoDescriptor(DefaultMavenPluginManager.java:286)
	at org.apache.maven.plugin.DefaultBuildPluginManager.getMojoDescriptor(DefaultBuildPluginManager.java:244)
	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.setupMojoExecution(DefaultLifecycleExecutionPlanCalculator.java:169)
	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.setupMojoExecutions(DefaultLifecycleExecutionPlanCalculator.java:154)
	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateExecutionPlan(DefaultLifecycleExecutionPlanCalculator.java:130)
	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateExecutionPlan(DefaultLifecycleExecutionPlanCalculator.java:144)
	at org.apache.maven.lifecycle.internal.builder.BuilderCommon.resolveBuildPlan(BuilderCommon.java:97)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:111)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Sending interrupt signal to process
Killing processes
kill finished with exit code 0
```

This seems like a apache httpclient version mismatch somewhere.
Attempting to identify why this happens.

Reply:
The bad httpclient version when it fails is version 4.0.5. (which doesn't have the URIUtil.extractHost method).

Looks like google-cloud (transitive dep), test-distribution (transitive dep), anything that uses maven-dependency-plugin (as plugin dep), and errorprone profile all bring in different versions of apache httpclient.

Reply:
I opened issue #5924 for this specific issue.

I need to get the other parts of this PR merged now.